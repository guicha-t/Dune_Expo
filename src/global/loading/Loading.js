import React, { Component } from 'react';
import {View, StyleSheet, ActivityIndicator} from 'react-native';
import {
    BallIndicator,
    BarIndicator,
    DotIndicator,
    MaterialIndicator,
    PacmanIndicator,
    PulseIndicator,
    SkypeIndicator,
    UIActivityIndicator,
    WaveIndicator,
  } from 'react-native-indicators';

import { observer } from 'mobx-react';

import Header from './../../global/header/Header';
import Store from './../../global/store/Store';
import * as cfg from "./../../Config";

@observer
export default class Dashboard extends Component {
  render() {
    return(
      <View style={{flex:1}}>
        <View style={styles.loadingContainer}>
          <BarIndicator color={cfg.SECONDARY}/>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  loadingContainer: {
    flex: 1,
    backgroundColor: '#F9F9F9',
    justifyContent: 'center',
    alignItems: 'center',
  },
});
