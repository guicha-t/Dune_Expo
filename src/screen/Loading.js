import React, { Component } from 'react';
import {
  ActivityIndicator,
  AsyncStorage,
  StatusBar,
  StyleSheet,
  View,
} from 'react-native';

import Store from './../global/store/Store'

export default class AuthLoadingScreen extends Component {
  constructor(props) {
    super(props);
    this._bootstrapAsync();
  }

  _bootstrapAsync = async () => {
    const localId = await AsyncStorage.getItem('localId');
    const localToken = await AsyncStorage.getItem('localToken');
    const localType = await AsyncStorage.getItem('localType');


    if (localId) {
      Store.setIdUser(localId)
      Store.setToken(localToken)
      Store.setTypeUser(localType)
    }
    this.props.navigation.navigate(localId ? 'Dashboard' : 'Start');
  };

  render() {
    return (
      <View style={styles.container}>
        <ActivityIndicator />
        <StatusBar barStyle="default" />
      </View>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#FEE599',
  },
});
