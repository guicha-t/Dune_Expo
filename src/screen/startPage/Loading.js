import React, { Component } from 'react';
import {
  ActivityIndicator,
  AsyncStorage,
  StatusBar,
  StyleSheet,
  View,
  Alert,
} from 'react-native';

import {
    BallIndicator,
    BarIndicator,
    DotIndicator,
    MaterialIndicator,
    PacmanIndicator,
    PulseIndicator,
    SkypeIndicator,
    UIActivityIndicator,
    WaveIndicator,
  } from 'react-native-indicators';

import Store from './../../global/store/Store'
import Header from './../../global/header/Header';
import * as cfg from "./../../Config";

export default class AuthLoadingScreen extends Component {
  constructor(props) {
    super(props);
    this._bootstrapAsync();
  }

  async _removeItemValue(key) {
    try {
      await AsyncStorage.removeItem(key);
      return true;
    }
    catch(exception) {
      return false;
    }
  }


  _bootstrapAsync = async () => {
    const localId = await AsyncStorage.getItem('localId');
    const localToken = await AsyncStorage.getItem('localToken');
    const localType = await AsyncStorage.getItem('localType');

    fetch(cfg.API_URL + '/tokens/verifyToken', {
      method: 'POST',
      Accept: 'application/json',
      headers: {
        'Content-Type': 'application/json',
        token: Store.Token,
      },
      body: JSON.stringify({
        token: localToken,
      }),
    }).then((response) => response.json())
    .then((responseJson) => {
      if ((responseJson.response != 'Token valid') && (localToken))
      {
        this._removeItemValue("localToken")
        this._removeItemValue("localType")
        Store.setToken('');
        Store.setTypeUser('')
        Store.setIsLog(false);
        Alert.alert('Session expirée - Déconnexion');
        this.props.navigation.navigate('Start');
      }
      else {
        if (localToken) {
          Store.setToken(localToken)
          Store.setTypeUser(localType)
        }
        this.props.navigation.navigate(localToken ? 'Dashboard' : 'Start');
      }

      })
      .catch((error) => {
      console.error(error);
    });
  };

  render() {
    return (
      <View style={styles.container}>
        <BarIndicator color='#363453'/>
        <StatusBar barStyle="default" />
      </View>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#F9F9F9',
  },
});
