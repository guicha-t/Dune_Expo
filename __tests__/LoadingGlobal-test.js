import Loading from './../src/global/loading/Loading';
import 'isomorphic-fetch';
import React from 'react';

import renderer from 'react-test-renderer'

jest.useFakeTimers();

test('render Loading correctly', () => {

    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
});