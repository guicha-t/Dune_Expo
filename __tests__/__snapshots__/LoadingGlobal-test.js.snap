// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Loading correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#FFF",
        },
      ]
    }
  >
    <View
      animating={true}
      animationDuration={1200}
      animationEasing={[Function]}
      color="#2A5E85"
      interaction={true}
      renderComponent={[Function]}
      size={40}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "marginHorizontal": 4,
            "width": 8,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#2A5E85",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "height": 20,
              "transform": Array [
                Object {
                  "translateY": 8,
                },
              ],
              "width": 8,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#2A5E85",
              "borderBottomLeftRadius": 4,
              "borderBottomRightRadius": 4,
              "height": 20,
              "transform": Array [
                Object {
                  "translateY": -8,
                },
              ],
              "width": 8,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "height": 40,
            "marginHorizontal": 4,
            "width": 8,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#2A5E85",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "height": 20,
              "transform": Array [
                Object {
                  "translateY": 4.898587196589413e-16,
                },
              ],
              "width": 8,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#2A5E85",
              "borderBottomLeftRadius": 4,
              "borderBottomRightRadius": 4,
              "height": 20,
              "transform": Array [
                Object {
                  "translateY": -4.898587196589413e-16,
                },
              ],
              "width": 8,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "height": 40,
            "marginHorizontal": 4,
            "width": 8,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#2A5E85",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
              "height": 20,
              "transform": Array [
                Object {
                  "translateY": 8,
                },
              ],
              "width": 8,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#2A5E85",
              "borderBottomLeftRadius": 4,
              "borderBottomRightRadius": 4,
              "height": 20,
              "transform": Array [
                Object {
                  "translateY": -8,
                },
              ],
              "width": 8,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
